### ISSUE TITLE
When generate docs, it throws errors, like "error TS2304: Cannot find name 'Iterable'". Is there any version problems?

### ISSUE THREAD

**Issue by JuneScut** [TypeStrong/typedoc#2655]
## Search terms

version, ts error

## Expected Behavior

Genrate docs correctly

## Actual Behavior

throw ts error, and failed to generate docs

## Steps to reproduce the bug

<img width="389" alt="image" src="https://github.com/user-attachments/assets/4cc36ca0-3e14-4c61-96bb-9d9c54a9ad17">
<img width="1041" alt="image" src="https://github.com/user-attachments/assets/a6ae91f7-bd01-42b2-a798-3eb1ecfdb6c1">

## Environment

-   Typedoc version: 0.25.13
-   TypeScript version: 4.9.5
-   Node.js version: v18.20.0
-   OS: M2 Pro, macOS  sonoma 14.5

**Comment by Gerrit0** [comment]
That setCompilerOptions call looks incredibly suspicious. Spreading `app.options` won't give you valid compiler options.

**Comment by JuneScut** [comment]
It's a big repo. I don't want to generate docs for so many files. setCompilerOptions is a simple way to receive it. Using entryPoints or exclude cannot actually exclude files for generating docs.

Besides, when I use lower version of typedoc, I can generate docs except full categories.

**Comment by JuneScut** [comment]
If you close the issue, please give me a valid way to generate doc for specific file.

**Comment by Gerrit0** [comment]
If you're looking for support you:

1. Should provide a reproducible example. Images are the worst possible way to share information as any code within them cannot be copied by anyone trying to help.
2. Should use TypeScript, or at least add `// @ts-check` so TypeScript can report problems
3. Should describe your problem, and what you've tried. The original issue made no mention that you migrated from an older TypeDoc version, and while I *could* potentially have inferred that from code in your images, I make a point to not spend much time on low effort requests. I stopped reading that code as soon as I noticed one thing wrong with it.

---

There are problems due you your misuse of the API. If you use [getCompilerOptions](https://typedoc.org/api/classes/Configuration.Options.html#getCompilerOptions) instead of spreading `app.options`, you might have a chance of that working.
